conjure-core/src/main/java/com/palantir/conjure/parser/types/names/FieldName.java

```diff
@@ -38,60 +37,30 @@
 
     private static final Logger log = LoggerFactory.getLogger(FieldName.class);
 
-    private static final Pattern CAMEL_CASE_PATTERN =
-            Pattern.compile("^[a-z]([A-Z]{1,2}[a-z0-9]|[a-z0-9])*[A-Z]?$");
-    private static final Pattern KEBAB_CASE_PATTERN =
-            Pattern.compile("^[a-z]((-[a-z]){1,2}[a-z0-9]|[a-z0-9])*(-[a-z])?$");
-    private static final Pattern SNAKE_CASE_PATTERN =
-            Pattern.compile("^[a-z]((_[a-z]){1,2}[a-z0-9]|[a-z0-9])*(_[a-z])?$");
-
-    public enum Case {
-        LOWER_CAMEL_CASE(CAMEL_CASE_PATTERN, CaseFormat.LOWER_CAMEL),
-        KEBAB_CASE(KEBAB_CASE_PATTERN, CaseFormat.LOWER_HYPHEN),
-        SNAKE_CASE(SNAKE_CASE_PATTERN, CaseFormat.LOWER_UNDERSCORE);
-
-        private final Pattern pattern;
-        private final CaseFormat format;
-
-        Case(Pattern pattern, CaseFormat format) {
-            this.pattern = pattern;
-            this.format = format;
-        }
-
-        @Override
-        public String toString() {
-            return name() + "[" + pattern + "]";
-        }
-
-        public String convertTo(FieldName fieldName, Case targetCase) {
-            return format.to(targetCase.format, fieldName.name());
-        }
-    }
-
     @JsonValue
     public abstract String name();
 
     @Value.Check
     @SuppressWarnings("Slf4jLogsafeArgs")
     protected final void check() {
         Preconditions.checkArgument(
-                Case.LOWER_CAMEL_CASE.pattern.matcher(name()).matches()
-                        || Case.KEBAB_CASE.pattern.matcher(name()).matches()
-                        || Case.SNAKE_CASE.pattern.matcher(name()).matches(),
+                CaseConverter.Case.LOWER_CAMEL_CASE.getPattern().matcher(name()).matches()
+                        || CaseConverter.Case.KEBAB_CASE.getPattern().matcher(name()).matches()
+                        || CaseConverter.Case.SNAKE_CASE.getPattern().matcher(name()).matches(),
                 "FieldName \"%s\" must follow one of the following patterns: %s",
-                name(), Arrays.toString(Case.values()));
+                name(), Arrays.toString(CaseConverter.Case.values()));
```

Comment:
This is a bit dodgy that we're listing them explicily up there, but grabbing the list from CaseConverter here.
Can we have one or the other? Either re-define the list as a private static here, or always use `CaseConverter.Case.values()`